### PR TITLE
chore: deprecate SetParamValue/GetParamValue/Serialize Set/Pos/Rotation/Scale for IAnimNode

### PR DESCRIPTION
**Pull request by pollend**
issue: https://github.com/o3de/o3de/issues/9326

this is a continuation of: https://github.com/o3de/o3de/pull/9134

Signed-off-by: Michael Pollind <mpollind@gmail.com>

### CODE REVIEW

**Comment by cgalvan** [comment]
Starting an AR build for this now

**Review by hultonha — APPROVED** [review]
LGTM! Nice work @pollend 👍

**Comment by pollend** [comment]
> LGTM! Nice work @pollend +1

wondering about going back through after this and swapping endpoints to AZ Vector equivalent.

**Comment by hultonha** [comment]
> > LGTM! Nice work @pollend +1
> 
> wondering about going back through after this and swapping endpoints to AZ Vector equivalent.

Yeah sounds like a good plan, I'd do that as a separate PR 👍

File: Code/Legacy/CryCommon/IMovieSystem.h [hunk 1]
```diff
@@ -14,6 +14,10 @@
 #include <AzCore/Math/Quaternion.h>
 #include <AzCore/Serialization/SerializeContext.h>
 
+#include <AzCore/Math/Vector2.h>
```
Comment (AMZN-alexpete) [inline Code/Legacy/CryCommon/IMovieSystem.h:17]: You can likely remove this include if Vector2 is not actually used.

**Review by AMZN-alexpete — APPROVED** [review]
Just one minor suggestion

**Comment by cgalvan** [comment]
@pollend Can you please fix these compile issues?

```
ComponentEntityEditorPlugin.vcxproj -> D:\workspace\o3de\build\windows\bin\release\EditorPlugins\ComponentEntityEditorPlugin.dll

[2022-05-03T18:54:11.016Z]   unity_4_cxx.cxx

[2022-05-03T18:54:11.016Z] D:\workspace\o3de\Gems\Maestro\Code\Source\Cinematics\AnimNode.h(215,1): error C4266: 'bool IAnimNode::GetParamValue(float,CAnimParamType,AZ::Vector4 &)': no override available for virtual member function from base 'IAnimNode'; function is hidden [D:\workspace\o3de\build\windows\External\Maestro-38d5c635\Code\Maestro.Static.vcxproj]

[2022-05-03T18:54:11.016Z]   };

[2022-05-03T18:54:11.016Z]   ^

[2022-05-03T18:54:11.016Z]   D:\workspace\o3de\Code\Legacy\CryCommon\.\IMovieSystem.h(779): note: see declaration of 'IAnimNode::GetParamValue'

[2022-05-03T18:54:11.016Z]       virtual bool GetParamValue(float time, CAnimParamType param, AZ::Vector4& value) 

[2022-05-03T18:54:11.016Z]   D:\workspace\o3de\Code\Legacy\CryCommon\.\IShader.h(92): note: see declaration of 'IAnimNode'

[2022-05-03T18:54:11.016Z]   struct IAnimNode;

[2022-05-03T18:54:11.016Z] D:\workspace\o3de\Gems\Maestro\Code\Source\Cinematics\AnimNode.h(215,1): error C4266: 'bool IAnimNode::GetParamValue(float,CAnimParamType,AZ::Vector3 &)': no override available for virtual member function from base 'IAnimNode'; function is hidden [D:\workspace\o3de\build\windows\External\Maestro-38d5c635\Code\Maestro.Static.vcxproj]

[2022-05-03T18:54:11.016Z]   };

[2022-05-03T18:54:11.016Z]   ^

[2022-05-03T18:54:11.016Z] D:\workspace\o3de\Gems\Maestro\Code\Source\Cinematics\AnimNode.h(215,1): error C4266: 'bool IAnimNode::GetParamValue(float,CAnimParamType,AZ::Vector4 &)': no override available for virtual member function from base 'IAnimNode'; function is hidden [D:\workspace\o3de\build\windows\External\Maestro-38d5c635\Code\Maestro.Static.vcxproj]

[2022-05-03T18:54:11.016Z] D:\workspace\o3de\Gems\Maestro\Code\Source\Cinematics\AnimNode.h(215,1): error C4266: }; [D:\workspace\o3de\build\windows\External\Maestro-38d5c635\Code\Maestro.Static.vcxproj]

[2022-05-03T18:54:11.016Z] D:\workspace\o3de\Gems\Maestro\Code\Source\Cinematics\AnimNode.h(215,1): error C4266: ^ [D:\workspace\o3de\build\windows\External\Maestro-38d5c635\Code\Maestro.Static.vcxproj]

[2022-05-03T18:54:11.016Z]   D:\workspace\o3de\Code\Legacy\CryCommon\.\IMovieSystem.h(767): note: see declaration of 'IAnimNode::GetParamValue'

[2022-05-03T18:54:11.016Z]       virtual bool GetParamValue(float time, CAnimParamType param, AZ::Vector3& value) 

[2022-05-03T18:54:11.016Z]   D:\workspace\o3de\Code\Legacy\CryCommon\.\IShader.h(92): note: see declaration of 'IAnimNode'

[2022-05-03T18:54:11.016Z]   struct IAnimNode;

[2022-05-03T18:54:11.016Z] D:\workspace\o3de\Code\Legacy\CryCommon\.\IMovieSystem.h(779): message : see declaration of 'IAnimNode::GetParamValue' [D:\workspace\o3de\build\windows\External\Maestro-38d5c635\Code\Maestro.Static.vcxproj]

[2022-05-03T18:54:11.016Z] D:\workspace\o3de\Code\Legacy\CryCommon\.\IMovieSystem.h(779): message :     virtual bool GetParamValue(float time, CAnimParamType param, AZ::Vector4& value) [D:\workspace\o3de\build\windows\External\Maestro-38d5c635\Code\Maestro.Static.vcxproj]

[2022-05-03T18:54:11.016Z] D:\workspace\o3de\Code\Legacy\CryCommon\.\IShader.h(92): message : see declaration of 'IAnimNode' [D:\workspace\o3de\build\windows\External\Maestro-38d5c635\Code\Maestro.Static.vcxproj]

[2022-05-03T18:54:11.016Z] D:\workspace\o3de\Code\Legacy\CryCommon\.\IShader.h(92): message : struct IAnimNode; [D:\workspace\o3de\build\windows\External\Maestro-38d5c635\Code\Maestro.Static.vcxproj]

[2022-05-03T18:54:11.016Z] D:\workspace\o3de\Gems\Maestro\Code\Source\Cinematics\AnimNode.h(215,1): error C4266: 'bool IAnimNode::GetParamValue(float,CAnimParamType,AZ::Vector3 &)': no override available for virtual member function from base 'IAnimNode'; function is hidden [D:\workspace\o3de\build\windows\External\Maestro-38d5c635\Code\Maestro.Static.vcxproj]

[2022-05-03T18:54:11.016Z] D:\workspace\o3de\Gems\Maestro\Code\Source\Cinematics\AnimNode.h(215,1): error C4266: }; [D:\workspace\o3de\build\windows\External\Maestro-38d5c635\Code\Maestro.Static.vcxproj]

[2022-05-03T18:54:11.016Z] D:\workspace\o3de\Gems\Maestro\Code\Source\Cinematics\AnimNode.h(215,1): error C4266: ^ [D:\workspace\o3de\build\windows\External\Maestro-38d5c635\Code\Maestro.Static.vcxproj]

[2022-05-03T18:54:11.016Z] D:\workspace\o3de\Gems\Maestro\Code\Source\Cinematics\AnimNode.h(215,1): error C4266: 'bool IAnimNode::GetParamValue(float,CAnimParamType,AZ::Vector4 &)': no override available for virtual member function from base 'IAnimNode'; function is hidden [D:\workspace\o3de\build\windows\External\Maestro-38d5c635\Code\Maestro.Static.vcxproj]

[2022-05-03T18:54:11.016Z]   };

[2022-05-03T18:54:11.016Z]   ^

[2022-05-03T18:54:11.016Z]   D:\workspace\o3de\Code\Legacy\CryCommon\.\IMovieSystem.h(779): note: see declaration of 'IAnimNode::GetParamValue'

[2022-05-03T18:54:11.016Z] D:\workspace\o3de\Code\Legacy\CryCommon\.\IMovieSystem.h(767): message : see declaration of 'IAnimNode::GetParamValue' [D:\workspace\o3de\build\windows\External\Maestro-38d5c635\Code\Maestro.Static.vcxproj]

[2022-05-03T18:54:11.016Z] D:\workspace\o3de\Code\Legacy\CryCommon\.\IMovieSystem.h(767): message :     virtual bool GetParamValue(float time, CAnimParamType param, AZ::Vector3& value) [D:\workspace\o3de\build\windows\External\Maestro-38d5c635\Code\Maestro.Static.vcxproj]

[2022-05-03T18:54:11.016Z]       virtual bool GetParamValue(float time, CAnimParamType param, AZ::Vector4& value) 

[2022-05-03T18:54:11.016Z]   D:\workspace\o3de\Code\Legacy\CryCommon\.\IShader.h(92): note: see declaration of 'IAnimNode'

[2022-05-03T18:54:11.016Z]   struct IAnimNode;

[2022-05-03T18:54:11.016Z] D:\workspace\o3de\Gems\Maestro\Code\Source\Cinematics\AnimNode.h(215,1): error C4266: 'bool IAnimNode::GetParamValue(float,CAnimParamType,AZ::Vector3 &)': no override available for virtual member function from base 'IAnimNode'; function is hidden [D:\workspace\o3de\build\windows\External\Maestro-38d5c635\Code\Maestro.Static.vcxproj]

[2022-05-03T18:54:11.016Z]   };

[2022-05-03T18:54:11.016Z]   ^

[2022-05-03T18:54:11.017Z]   D:\workspace\o3de\Code\Legacy\CryCommon\.\IMovieSystem.h(767): note: see declaration of 'IAnimNode::GetParamValue'

[2022-05-03T18:54:11.017Z]       virtual bool GetParamValue(float time, CAnimParamType param, AZ::Vector3& value) 

[2022-05-03T18:54:11.017Z]   D:\workspace\o3de\Code\Legacy\CryCommon\.\IShader.h(92): note: see declaration of 'IAnimNode'

[2022-05-03T18:54:11.017Z]   struct IAnimNode;

[2022-05-03T18:54:11.017Z] D:\workspace\o3de\Gems\Maestro\Code\Source\Cinematics\AnimNode.h(215,1): error C4266: 'bool IAnimNode::GetParamValue(float,CAnimParamType,AZ::Vector4 &)': no override available for virtual member function from base 'IAnimNode'; function is hidden [D:\workspace\o3de\build\windows\External\Maestro-38d5c635\Code\Maestro.Static.vcxproj]

[2022-05-03T18:54:11.017Z]   };

[2022-05-03T18:54:11.017Z]   ^

[2022-05-03T18:54:11.017Z]   D:\workspace\o3de\Code\Legacy\CryCommon\.\IMovieSystem.h(779): note: see declaration of 'IAnimNode::GetParamValue'

[2022-05-03T18:54:11.017Z]       virtual bool GetParamValue(float time, CAnimParamType param, AZ::Vector4& value) 

[2022-05-03T18:54:11.017Z]   D:\workspace\o3de\Code\Legacy\CryCommon\.\IShader.h(92): note: see declaration of 'IAnimNode'

[2022-05-03T18:54:11.017Z]   struct IAnimNode;

[2022-05-03T18:54:11.017Z] D:\workspace\o3de\Gems\Maestro\Code\Source\Cinematics\AnimNode.h(215,1): error C4266: 'bool IAnimNode::GetParamValue(float,CAnimParamType,AZ::Vector3 &)': no override available for virtual member function from base 'IAnimNode'; function is hidden [D:\workspace\o3de\build\windows\External\Maestro-38d5c635\Code\Maestro.Static.vcxproj]

[2022-05-03T18:54:11.017Z]   };

[2022-05-03T18:54:11.017Z]   ^

[2022-05-03T18:54:11.017Z]   D:\workspace\o3de\Code\Legacy\CryCommon\.\IMovieSystem.h(767): note: see declaration of 'IAnimNode::GetParamValue'

[2022-05-03T18:54:11.017Z]       virtual bool GetParamValue(float time, CAnimParamType param, AZ::Vector3& value) 

[2022-05-03T18:54:11.017Z]   D:\workspace\o3de\Code\Legacy\CryCommon\.\IShader.h(92): note: see declaration of 'IAnimNode'

[2022-05-03T18:54:11.017Z]   struct IAnimNode;

[2022-05-03T18:54:11.280Z]   EMotionFX.Editor.Static.vcxproj -> D:\workspace\o3de\build\windows\lib\release\EMotionFX.Editor.Static.lib

[2022-05-03T18:54:11.280Z]   PhysX.Editor.Tests.vcxproj -> D:\workspace\o3de\build\windows\bin\release\PhysX.Editor.Tests.Gem.dll

[2022-05-03T18:54:11.280Z]   Automatic MOC for target ProjectSettingsTool

[2022-05-03T18:54:11.542Z]   ProjectSettingsTool.vcxproj -> D:\workspace\o3de\build\windows\bin\release\EditorPlugins\ProjectSettingsTool.dll

[2022-05-03T18:54:11.542Z]   Automatic MOC for target EMotionFX.Editor.Tests

[2022-05-03T18:54:11.542Z]   EMotionFX.Editor.vcxproj -> D:\workspace\o3de\build\windows\bin\release\EMotionFX.Editor.dll

[2022-05-03T18:54:11.804Z]   EMotionFX.Editor.Tests.vcxproj -> D:\workspace\o3de\build\windows\bin\release\EMotionFX.Editor.Tests.dll

[2022-05-03T18:54:11.804Z]   NvCloth.Editor.Tests.vcxproj -> D:\workspace\o3de\build\windows\bin\release\NvCloth.Editor.Tests.Gem.dll

[2022-05-03T18:54:11.804Z]   EMotionFX_Atom.Editor.vcxproj -> D:\workspace\o3de\build\windows\bin\release\EMotionFX_Atom.Editor.dll

[2022-05-03T18:54:11.804Z]   Atom_AtomBridge.Editor.vcxproj -> D:\workspace\o3de\build\windows\bin\release\Atom_AtomBridge.Editor.dll

[2022-05-03T18:54:16.034Z] D:\workspace\o3de\Code\Legacy\CryCommon\.\IShader.h(92): message : see declaration of 'IAnimNode' [D:\workspace\o3de\build\windows\External\Maestro-38d5c635\Code\Maestro.Static.vcxproj]

[2022-05-03T18:54:16.035Z] D:\workspace\o3de\Code\Legacy\CryCommon\.\IShader.h(92): message : struct IAnimNode; [D:\workspace\o3de\build\windows\External\Maestro-38d5c635\Code\Maestro.Static.vcxproj]
```

**Comment by pollend** [comment]
opps wasn't suppose to make those virtual. and I removed the duplicate headers.

**Comment by hultonha** [comment]
I just kicked off another build, thanks for updating :+1:

**Comment by hultonha** [comment]
Hey @pollend, we identified one more flaky Physics test and have temporarily disabled it here - https://github.com/o3de/o3de/pull/9423. Once that is merged into development, can you update your branch and then I'll kick off another build 👍 Thanks!

**Comment by cgalvan** [comment]
Thanks @pollend  I've kicked off a new AR run

**Comment by hultonha** [comment]
Hooray! 😄 Glad this finally got merged, thanks @cgalvan and @pollend! 🙂

**Pull request merged**